deepchem/models/keras_model.py

```diff
@@ -421,13 +433,19 @@ def fit_generator(self,
       avg_loss = float(avg_loss) / averaged_batches
       logger.info(
           'Ending global_step %d: Average loss %g' % (current_step, avg_loss))
+      avg_losses.append(avg_loss)
```

Comment:
The implementation is a bit more complicated than it really needs to be.  The only required change to this method is a single addition here:

```python
if all_losses is not None:
  all_losses.append(avg_loss)
```

Everything else can stay exactly as it was before.

Reply:
Good suggestion! I've simplified it in this fashion, with a minor tweak to handle the edge case that was failing in #1944.